src/herder/Upgrades.cpp

```diff
@@ -1466,42 +1466,122 @@ ConfigUpgradeSetFrame::isValidForApply() const
+                    cfg.contractBandwidth().ledgerMaxPropagateSizeBytes >
+                        MinimumSorobanNetworkConfig::MIN_TX_SIZE_BYTES &&
+                    cfg.contractBandwidth().txMaxSizeBytes >=
+                        MinimumSorobanNetworkConfig::MIN_TX_SIZE_BYTES;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_COMPUTE_V0:
+            valid = cfg.contractCompute().feeRatePerInstructionsIncrement > 0 &&
+                    cfg.contractCompute().ledgerMaxInstructions >=
+                        MinimumSorobanNetworkConfig::MIN_INSTRUCTIONS &&
+                    cfg.contractCompute().txMaxInstructions >=
+                        MinimumSorobanNetworkConfig::MIN_INSTRUCTIONS &&
+                    cfg.contractCompute().txMemoryLimit >=
+                        MinimumSorobanNetworkConfig::MIN_MEMORY;
+
+            valid = valid && cfg.contractCompute().ledgerMaxInstructions >=
+                                 cfg.contractCompute().txMaxInstructions;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_HISTORICAL_DATA_V0:
+            valid = cfg.contractHistoricalData().feeHistorical1KB > 0;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_LEDGER_COST_V0:
+            valid = cfg.contractLedgerCost().ledgerMaxReadLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_READ_ENTRIES &&
+                    cfg.contractLedgerCost().ledgerMaxReadBytes >
+                        MinimumSorobanNetworkConfig::MIN_READ_BYTES &&
+                    cfg.contractLedgerCost().ledgerMaxWriteLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_ENTRIES &&
+                    cfg.contractLedgerCost().ledgerMaxWriteBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractLedgerCost().txMaxReadLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_READ_ENTRIES &&
+                    cfg.contractLedgerCost().txMaxReadBytes >
+                        MinimumSorobanNetworkConfig::MIN_READ_BYTES &&
+                    cfg.contractLedgerCost().txMaxWriteLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_ENTRIES &&
+                    cfg.contractLedgerCost().txMaxWriteBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractLedgerCost().feeReadLedgerEntry > 0 &&
+                    cfg.contractLedgerCost().feeWriteLedgerEntry > 0 &&
+                    cfg.contractLedgerCost().feeRead1KB > 0 &&
+                    cfg.contractLedgerCost().bucketListTargetSizeBytes > 0 &&
+                    cfg.contractLedgerCost().writeFee1KBBucketListLow > 0 &&
+                    cfg.contractLedgerCost().writeFee1KBBucketListHigh > 0 &&
+                    cfg.contractLedgerCost().bucketListWriteFeeGrowthFactor > 0;
```

Comment:
`bucketListWriteFeeGrowthFactor` should be allowed to be `0` (and that's what we'll probably start with). In general, I would allow every fee to be 0 just in case.